sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -62,12 +62,17 @@ export class BatchingReceiver extends MessageReceiver {
   /**
    * Receives a batch of messages from a ServiceBus Queue/Topic.
    * @param maxMessageCount      The maximum number of messages to receive.
+   * This has to be a number <= 2047 for receiver in `peekLock` mode, due to memory size
+   * constraints of underlying buffer.
+   * If a value > 2047 is used, then only 2047 messages are fetched at a time.
    * @param idleTimeoutInSeconds The maximum wait time in seconds for which the Receiver
    * should wait to receive the first message. If no message is received by this time,
    * the returned promise gets resolved to an empty array.
    * @returns {Promise<ServiceBusMessage[]>} A promise that resolves with an array of Message objects.
    */
   receive(maxMessageCount: number, idleTimeoutInSeconds?: number): Promise<ServiceBusMessage[]> {
+    maxMessageCount = maxMessageCount > 2047 ? 2047 : maxMessageCount;
```

Comment:
We have to do this only in PeekLock mode, because as you have found bigger numbers affect only PeekLock mode

Reply:
See https://github.com/Azure/azure-sdk-for-js/issues/4748#issuecomment-545670191 bigger numbers do affect receiveAndDelete mode as well.
Documentation and code can be updated once that is fixed, unless we want to regress to supporting only peeklock mode in batching receiver?

Reply:
I have reviewed the discussion in #4748 and after going through all the comments, it looks like the only change we need is to update the documentation.

Is there a specific need to set the count to 2047 here?

I know that initially we thought this could be the solution to the program hanging, but it was concluded in #4748 that the program doesnt hang, and messages are indeed returned with the limit at 2047.

Reply:
Since the receiver has a known issue in receiveAndDelete mode and that there is a possibility it may pull in more messages if the buffer gets freed up, setting the value to 2047 would help prevent any unintended side effects. So basically, if we know that only 2047 messages can and should be returned, might as well do it the clean way.

Reply:
Agree with fix applying for peekLock mode only - even if receiveAndDelete mode issue gets fixed, the limits would be different for it - updated to factor this in

Reply:
I would re-iterate my point from https://github.com/Azure/azure-sdk-for-js/pull/5758#discussion_r340380141. I believe only a documentation update is needed here. We don't have to do the code change of capping `maxMessageCount` to 2047.

If `rhea` increases this number tomorrow, then a customer will have to wait for us to ship an update to make use of the increased buffer size. Or if a customer has patched rhea to use a larger buffer, they will be forced to patch our library to make use of the update as well.

Reply:
Updated..